racket/src/io/network/tcp-port.rkt

```diff
@@ -92,3 +93,22 @@
     [(tcp-output-port? cp)
      (set-tcp-output-port-abandon?! cp #t)
      (close-port p)]))
+
+(define/who (tcp-sendfile dst-p src-p [offset 0] [nbytes 0])
+  (define op (->core-output-port dst-p #:default #f))
+  (unless (tcp-output-port? op)
+    (raise-argument-error who "(and/c tcp-port? output-port?)" dst-p))
+  (define ip (->core-input-port src-p))
+  (unless (file-stream-port? ip)
+    (raise-argument-error who "(and/c file-stream-port? input-port?)" src-p))
+  (unless (>= offset 0)
```

Comment:
This doesn't check that `offset` is an exact integer. Probably it's best to use `exact-nonnegative-integer?` here. Even better, convert to use the `check` form. Also, you'll need to do something about bignums before trying to send them to `rktio_sendfile` — maybe looping over fixnum-sized chunks or only writing a fixnum-sized check and letting the caller loop.